utils/src/lib.rs

```diff
@@ -29,6 +29,53 @@ pub fn change_endianness<const SIZE: usize>(bytes: &[u8; SIZE]) -> [u8; SIZE] {
     arr
 }
 
+/// Converts the given [`ark_ff::BigInt`](ark_ff::BigInt) into a big-endian
+/// byte array.
+pub fn bigint_to_be_bytes<const BYTES_SIZE: usize, const NUM_LIMBS: usize>(
```

Comment:
why don't you use the native ark ff BigInteger methods to_bytes_be and to_bytes_le?
https://docs.rs/ark-ff/latest/src/ark_ff/biginteger/mod.rs.html#448-461

Reply:
Because my function allocated an array with known size only once and writes limbs directly to it.

`to_bytes_be` allocates and writes to a vector, so then we would have to copy it again to an array (with `try_into()`). Which is unnecessary, because we know the size of the specific BigInt (through `NUM_LIBS` before).

I'm surprised that arkworks doesn't provide such method.

Reply:
Actually, thanks for pointing out to this code. I looked at it again and I have one more optimization in mind. :smiling_imp:  Let me push it.

Reply:
Done.

Reply:
nice! We should upstream at some point.